external_ml/seviri/predictCPHCOT.py

```diff
@@ -0,0 +1,377 @@
+    
+    First author: Daniel Philipp (DWD)
+    ---------------------------------------------------------------------------
+    2020/07/20, DP: Initial version
+    2020/07/23, DP: Added a neural network driver file containing the
+                    backend, paths and thresholds
+    2020/08/13, DP: Added masking of invalid (space) pixels before prediction.
+                    Only valid pixels are predicted to increase efficiency.
+                    Implemented correct fill value usage.
+    2020/08/18, DP: Implemented fully working uncertainty estimation.
+"""
+
+import neuralnet
+import xarray as xr
+import numpy as np
+import readdriver
+import os
+import time
+import logging
+import helperfuncs as hf
+from definitions import (SREAL_FILL_VALUE, BYTE_FILL_VALUE, SREAL,
+                         IS_CLEAR, IS_CLOUD, IS_WATER, IS_ICE)
+#import tensorflow as tf
+import theano
+
+fmt = '%(levelname)s : %(filename)s : %(message)s'
+logging.basicConfig(level=logging.DEBUG,
+                    format=fmt)
+
+theano.gof.compilelock.set_lock_status(False)
+
+def _prepare_input_arrays(vis006, vis008, nir016, ir039, ir062, ir073, ir087,
+				      ir108, ir120, ir134, lsm, skt, networks):
+    """ 
+        Prepare input array for the neural network. Takes required feature 
+        arrays, flattens them using row-major ordering and combines all flat 
+        arrays into a single array with shape (nsamples, nfeatures) to be 
+        used for prediction.
+        
+        Input:
+        - vis006 (2d numpy array): SEVIRI VIS 0.6 um (Ch 1)
+        - vis008 (2d numpy array): SEVIRI VIS 0.8 um (Ch 2)
+        - nir016 (2d numpy array): SEVIRI NIR 1.6 um (Ch 3)
+        - ir039 (2d numpy array):  SEVIRI IR 3.9 um  (Ch 4)
+        - ir062 (2d numpy array):  SEVIRI WV 6.2 um  (Ch 5)
+        - ir073 (2d numpy array):  SEVIRI WV 7.3 um  (Ch 6)
+        - ir087 (2d numpy array):  SEVIRI IR 8.7 um  (Ch 7)
+        - ir108 (2d numpy array):  SEVIRI IR 10.8 um (Ch 9)
+        - ir120 (2d numpy array):  SEVIRI IR 12.0 um (Ch 10)
+        - ir134 (2d numpy array):  SEVIRI IR 13.4 um  (Ch 11)
+        - lsm (2d numpy array):    Land-sea mask
+        - skt (2d numpy array):    (ERA5) Skin Temperature
+        
+        Return:
+        - idata (2d numpy array): Scaled input array for ANN 
+    """
+    
+    # Change VIS/IR range from [0,1] to [0,100] to match trained network
+    vis006 = np.where(vis006 > SREAL_FILL_VALUE, vis006 * 100, vis006)
+    vis008 = np.where(vis008 > SREAL_FILL_VALUE, vis008 * 100, vis008)
+    nir016 = np.where(nir016 > SREAL_FILL_VALUE, nir016 * 100, nir016)
+
+    # calculate channel differences
+    ir087_108 = ir087 - ir108
+    ir108_120 = ir108 - ir120
+    
+    # list of arrays must be kept in this order!
+    data_lst = [
+                ir039,     # 1    
+                ir087,     # 2   
+                ir087_108, # 3   
+                ir108,     # 4    
+                ir108_120, # 5   
```

Comment:
At least two spaces required for inline comment.

Reply:
Corrected with latest commit 0e2113b.